### PR TITLE
github/workflows/editorconfig: add check for nixpkgs-fmt

### PR DESCRIPTION
**Pull request by Synthetica9**
**DO NOT MERGE**

This should be held off until nixpkgs has been globally formatted with nixpkgs-fmt (#121490)

Adds a check for nixpkgs-fmt, which seems destined to become the new game in town when it comes to formatting nixpkgs.

Example PRs on my own fork:

* Failing editorconfig: https://github.com/Synthetica9/nixpkgs/pull/6
* Failing nixpkgs-fmt: https://github.com/Synthetica9/nixpkgs/pull/7
* Passing both: https://github.com/Synthetica9/nixpkgs/pull/8

cc @domenkozar 

Related: #120832 #121490 #121379

### CODE REVIEW

File: .github/workflows/editorconfig.yml [hunk 1]
```diff
@@ -33,14 +33,27 @@ jobs:
...
-
+    - name: Checking nixpkgs-fmt
+      id: nixpkgs-fmt
+      continue-on-error: true
+      if: env.PR_DIFF
+      run: |
+        echo "$PR_DIFF" | jq '.' -r | grep '.nix' | xargs -rd '\n' -- nixpkgs-fmt --check
+    - if: steps.nixpkgs-fmt.outcome == 'failure'
+      run: |
+        echo "::error :: Hey! It looks like your changes don't agree with nixpkgs-fmt."
```
Comment (SuperSandro2000) [inline .github/workflows/editorconfig.yml:56]:
```suggestion
        echo "::error :: Hey! It looks like your changes don't follow nixpkgs-fmt. Run nixpkgs-fmt on all .nix files to fix this check."
```
just an idea.

Comment (Synthetica9) [reply .github/workflows/editorconfig.yml]: Yeah, probably better

**Comment by Synthetica9** [comment]
(I left the check disabled for other owners, but this now works fine for other repos as well, so we could disable that if we wanted?)

**Comment by l0b0** [comment]
Could [Alejandra](https://github.com/kamadorueda/alejandra) be considered as an alternative? It seems to be much faster than `nixpkgs-fmt`, while producing (IMO, obviously) better-looking results.

**Comment by Artturin** [comment]
Reopen when https://github.com/NixOS/rfcs/pull/101 is merged

**Pull request closed without merging**